src/test/java/testCases/AccessControlTest.java

```diff
@@ -0,0 +1,96 @@
+package testCases;
+
+import Config.ExtentManager;
+import PageObjects.HomePage;
+import PageObjects.LoginPage;
+import com.aventstack.extentreports.ExtentTest;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.testng.Assert;
+import org.testng.ITestResult;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import Constant.Constant;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import java.time.Duration;
+
+public class AccessControlTest {
+
+    ExtentTest extentTest;
+    HomePage homePage;
+    LoginPage loginPage;
+
+    @BeforeMethod
+    public void beforeMethod() {
+        Constant.WEBDRIVER = new ChromeDriver();
+        Constant.WEBDRIVER.manage().window().maximize();
+
+        homePage = new HomePage();
+        loginPage = new LoginPage();
+    }
+
+    @AfterMethod
+    public void afterMethod(ITestResult result) {
+        if (result.getStatus() == ITestResult.FAILURE) {
+            extentTest.fail("Test failed: " + result.getThrowable().getMessage());
+            String screenshotPath = ExtentManager.captureScreenshot(Constant.WEBDRIVER, result.getName());
+            extentTest.addScreenCaptureFromPath(screenshotPath);
+        } else if (result.getStatus() == ITestResult.SUCCESS) {
+            extentTest.pass("Test passed");
+        } else if (result.getStatus() == ITestResult.SKIP) {
+            extentTest.skip("Test skipped: " + result.getThrowable().getMessage());
+        }
+        ExtentManager.flush();
+        Constant.WEBDRIVER.quit();
+    }
+
+    @Test
+    public void TC04() {
+        extentTest = ExtentManager.createTest("TC04", "Login page displays when unlogged user accesses protected page");
+        homePage.open();
+
+        Constant.WEBDRIVER.findElement(By.xpath("//a[@href='/Page/BookTicketPage.cshtml']")).click();
+
+        String currentUrl = Constant.WEBDRIVER.getCurrentUrl();
+        Assert.assertTrue(currentUrl.contains("Login"),
+                "User is not redirected to Login page when accessing Book Ticket without logging in");
+    }
+
+    @Test
+    public void TC06() {
+        extentTest = ExtentManager.createTest("TC06", "Additional pages display once user logged in");
```

Comment:
Use the description attribute of the @Test annotation from TestNG for the test case description.